monai/transforms/utils.py

```diff
@@ -625,9 +625,12 @@ def generate_label_classes_crop_centers(
 
     for i, array in enumerate(indices):
         if len(array) == 0:
-            ratios_[i] = 0
-            if warn:
-                warnings.warn(f"no available indices of class {i} to crop, set the crop ratio of this class to zero.")
+            if ratios_[i] != 0:
+                ratios_[i] = 0
+                if warn:
```

Comment:
I would suggest moving this warning outside of the previous if.

Reply:
Thanks for the feedback!

I'm not sure if I understand the suggestion. The proposal was about only setting the ratio for class `i`  to 0, if it is not already 0, and additionally, in case if `warn` is True, then print the warning message as well.

If I move the warning outside the previous if, like so:

```python
if len(array) == 0:
  if ratios_[i] != 0:
    ratios_[i] = 0
  if warn:
    warnings.warn(
      f"no available indices of class {i} to crop, setting the crop ratio of this class to zero.")
```
, then the warning will still get printed no matter if the ratio was already 0 for that class.

Reply:
Kindly disregard my previous comment, thanks!